event/filters.py

```diff
@@ -0,0 +1,120 @@
+from .models import Event, Like
+from user.models import OcialUser
+from ocial.models import Category
+from datetime import datetime
+from rest_framework.authtoken.models import Token
+from django.db.models.functions import ACos, Cos, Radians, Sin
+from django.db.models import F
+
+
+def filterEvents(request):
+    params = request.query_params
+    filters = params.keys() 
+    if ("timeStart" in filters and "timeEnd" in filters):
+        if params["timeStart"] > params["timeEnd"]:
+            return {
+                "error": "Invalid time",
+                "message": "timeStart must be before timeEnd",
+            }
+    if not (("longitude" in filters and "latitude" in filters and "radius" in filters) or (
+        "longitude" not in filters and "latitude" not in filters and "radius" not in filters
+    )):
```

Comment:
Prueba a ver si esto funciona igual, porque quede más limpio:

``if ["longitude", "latitude", "radius"] in filters``

Reply:
da un error de que list es unhasheable, asi que se queda como esta